main.py

```diff
@@ -0,0 +1,52 @@
+import random
+
```

Comment:
The code has good structure overall. Good use of functions to separate out the name generator and validation. I would add comments at most stages to explain what each group of lines specifically do and what they check for. This would make it more easily understandable to the reader.